### ISSUE TITLE
Low ESFM synth volume

### ISSUE THREAD

**Issue by Torinde** [joncampbell123/dosbox-x#5215]
### Question

@d-sphinx from https://github.com/dosbox-staging/dosbox-staging/issues/3911 says:
> When using new ESS sbtype the midi music (in-game) sounds very faint. About 4x times lower than normal OPL syth.
> - set sbtype = ess and configure the ESS Technology ES1688, ES1788, ES1888 Enhanced FM Audio MIDI music driver in the game > - setup utility
> - install and run any ESFM game (e.g. Heroes 2).
> - very weak music... The digital effects sound fine. The problem is that the midi music exists but barely audible.
> 
> My guess that ESS init procedure resets FM volume after game starts to some very low settings (like -13dB or so).
>
> A synthesizer drop of -13 dB is observed. This is independent of the game and its settings (as long as it is set to ES1688).
> ![2024-09-03 00_19_23-DOS - DOSBox-X 2024 03 01_ HEROES2 - 52000 cycles_ms PAUSED](https://private-user-images.githubusercontent.com/4455753/367882282-15ec1d91-ae7b-4064-96c1-054d14a8ac96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NDI0NjQsIm5iZiI6MTcyNjY0MjE2NCwicGF0aCI6Ii80NDU1NzUzLzM2Nzg4MjI4Mi0xNWVjMWQ5MS1hZTdiLTQwNjQtOTZjMS0wNTRkMTRhOGFjOTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MThUMDY0OTI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDYwMWIxOGM1ZGM3MDI1ZjFkYTM4YjZlNzAyYzc1NzViM2RjMGI0Yzg5ZDUyZjYwMjk4ZTI2ZWRmZTg1NWUyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.AEKtctw3j06zM6UfxuZByQMLjHmQr17pdTRJTZ_cvPA)

It seems he observes the same problem both in Staging and X.

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by johnnovak** [comment]
> It seems he observes the same problem both in Staging and X.

Yeah because I ported the code from X. I'm not gonna put work into this because it works well enough; just raise the OPL channel's volume. If this is an emulation bug and the original author fixes it (if...), then I'll port over the changes.

**Comment by kjliew** [comment]
This is caused by ESS mixer write `[reg 0x36, 0x88]` in the Miles Sound System ESFM driver. The driver reads the current mixer volume then writes 0x88. On game exits, the driver restores the mixer value from its previous read-back. 0x88 represents the midpoint of 4-bit left/right volume encoding. In-game MIDI music volume control does not touch ESS mixer.

There are 2 ways to interpret such observation:
1. The driver actually wanted softer volume hence selected a midpoint.
2. The 0x88 is indeed the 0-dB point for ESS mixer while Sound Blaster Pro 2.0 mixer has 0xFF as the 0-dB point.
